pkg/streamer/file_test.go

```diff
@@ -367,96 +381,37 @@ func (t *testUtilSuite) TestRelaySubDirUpdated(c *C) {
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		relaySubDirUpdated(ctx, watcherInterval, subDir, relayPaths[0], relayFiles[0], size, updatePathCh, errCh)
+		relayLogUpdatedOrNewCreated(ctx, watcherInterval, subDir, relayPaths[0], relayFiles[0], size, updatePathCh, errCh)
 		c.Assert(len(errCh), Equals, 1)
 		c.Assert(len(updatePathCh), Equals, 0)
 		err = <-errCh
 		c.Assert(err, ErrorMatches, ".*file size of relay log.*become smaller.*")
 	}()
 	wg.Wait()
 
-	// 3. new file created when adding watching
+	// 3. new binlog file created
 	err = ioutil.WriteFile(relayPaths[1], nil, 0o600)
 	c.Assert(err, IsNil)
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		relaySubDirUpdated(ctx, watcherInterval, subDir, relayPaths[0], relayFiles[0], 0, updatePathCh, errCh)
+		// update meta file
+		meta := Meta{BinLogName: relayFiles[1], BinLogPos: binlogPos, BinlogGTID: binlogGTID}
+		var buf bytes.Buffer
+		enc := toml.NewEncoder(&buf)
+		err2 := enc.Encode(meta)
+		c.Assert(err2, IsNil)
+		filename := path.Join(subDir, utils.MetaFilename)
+		err3 := utils.WriteFileAtomic(filename, buf.Bytes(), 0o644)
+		c.Assert(err3, IsNil)
+
+		relayLogUpdatedOrNewCreated(ctx, watcherInterval, subDir, relayPaths[0], relayFiles[0], 0, updatePathCh, errCh)
 		c.Assert(len(errCh), Equals, 0)
 		c.Assert(len(updatePathCh), Equals, 1)
 		up := <-updatePathCh
 		c.Assert(up, Equals, relayPaths[1])
 	}()
 	wg.Wait()
-
-	// 4. file updated
```

Comment:
why do you delete tests again?

Reply:
added new test case in https://github.com/pingcap/dm/pull/1856/commits/ae3415f7e887f076bc45c9255d7ebd317a9d5a5a